server/graphql/v2/mutation/CreateCollectiveMutations.js

```diff
@@ -0,0 +1,130 @@
+    settings: { ...DEFAULT_COLLECTIVE_SETTINGS },
+  };
+
+  const collectiveWithSlug = await models.Collective.findOne({ where: { slug: collectiveData.slug.toLowerCase() } });
+  if (collectiveWithSlug) {
+    throw new Error(`The slug ${collectiveData.slug} is already taken. Please use another slug for your collective.}.`);
+  }
+
+  let host;
+  if (args.host) {
+    host = fetchAccountWithInput(args.host, { loaders });
+    if (!host) {
+      throw new errors.ValidationFailed({ message: 'Host Not Found' });
+    }
+  }
+
+  // Handle GitHub automated approval for the Open Source Collective Host
+  if (args.automateApprovalWithGithub && args.githubHandle) {
+    const opensourceHost = defaultHostCollective('opensource');
+    if (host.id === opensourceHost.CollectiveId) {
+      try {
+        const githubAccount = await models.ConnectedAccount.findOne({
+          where: { CollectiveId: remoteUser.CollectiveId, service: 'github' },
+        });
+        if (!githubAccount) {
+          throw new Error('You must have a connected GitHub Account to claim a collective');
+        }
+        github.handleOpenSourceAutomatedApproval(args.githubHandle, githubAccount.token);
+        shouldAutomaticallyApprove = true;
+      } catch (error) {
+        throw new errors.ValidationFailed({
+          message: error.message,
+        });
+      }
+      if (args.githubHandle.includes('/')) {
+        collectiveData.settings.githubRepo = args.githubHandle;
+      } else {
+        collectiveData.settings.githubOrg = args.githubHandle;
+      }
+    }
+  }
+
+  const collective = await models.Collective.create(collectiveData);
+
+  // Add authenticated user as an admin
+  await collective.addUserWithRole(remoteUser, roles.ADMIN, { CreatedByUserId: remoteUser.id });
+
+  // Add the host if any
+  if (host) {
+    await collective.addHost(host, remoteUser, { shouldAutomaticallyApprove });
+    purgeCacheForPage(`/${host.slug}`);
+  }
+
+  // Will send an email to the authenticated user
+  // - tell them that their collective was successfully created
+  // - tell them that their collective is pending validation (which might be wrong if it was automatically approved)
+  const remoteUserCollective = await loaders.Collective.byId.load(remoteUser.CollectiveId);
+  models.Activity.create({
+    type: activities.COLLECTIVE_CREATED,
+    UserId: remoteUser.id,
+    CollectiveId: get(host, 'id'),
+    data: {
+      collective: collective.info,
+      host: get(host, 'info'),
+      user: {
+        email: remoteUser.email,
+        collective: remoteUserCollective.info,
+      },
+    },
+  });
+
+  return collective;
+}
+
+const createCollectiveMutations = {
+  createCollective: {
+    type: Collective,
+    args: {
+      collective: {
+        type: new GraphQLNonNull(CollectiveCreate),
+      },
+      host: {
+        type: AccountInput,
+      },
+      automateApprovalWithGithub: {
+        type: GraphQLBoolean,
+        defaultValue: false,
+      },
```

Comment:
@sbinlondon As top parameter, there is also `host` (ie: `{"slug": "opensource"}`) and `automateApprovalWithGithub` (boolean).

Reply:
And when would we automate approval with Github?

Reply:
Always when we pick the GitHub path.